src/driver-sinowealth.c

```diff
@@ -1003,9 +1694,10 @@ sinowealth_update_config_from_profile(struct ratbag_profile *profile)
 			config->glorious_mode = sinowealth_led_to_rgb_mode(led);
 			break;
 		case RATBAG_LED_BREATHING:
-			config->rgb_effect = RGB_BREATHING1;
-			config->breathing1_color = sinowealth_color_to_raw(device, led->color);
-			config->breathing1_mode = sinowealth_led_to_rgb_mode(led);
+			config->rgb_effect = RGB_BREATHING7;
```

Comment:
Won't this break the compatibility, maybe this should be a quirk of mouse Genesis Xenon 770 and DreamMachines DM5.

Also IIUC the case `RGB_BREATHING7` should be implemented for the function `sinowealth_update_profile_from_config` at https://github.com/libratbag/libratbag/blob/6ddbaa6c791e0d5dd120b9198a7411f47873cf5d/src/driver-sinowealth.c#L1045

Reply:
I don't think it would because the original author of this driver must have named the `breathing7_colorcount` for a reason (even though they said it's constant, I think they did so because there is no such option in the Glorious software). Also, gloryctl, a configuration utility for Model O, supports changing this field as well. 
Still, it would be really good if somebody with a Glorious mouse could test this, I can't as my mouse has no LEDs.

Regarding `sinowealth_update_profile_from_config`, you are right, I will fix it now.

Reply:
Thank you for pointing out [gloryctl](https://github.com/sammko/gloryctl), wasn't aware of that project.
So IIUC the RGB_BREATHING7 suppose to work on all glorious models, and the RGB_BREATHING1 work only on some?

Reply:
I think so, yes.